mackerel/api.go

```diff
@@ -170,7 +170,8 @@ func (api *API) FindHostByCustomIdentifier(customIdentifier string) (*Host, erro
 	}
 
 	if len(data.Hosts) == 0 {
-		return nil, fmt.Errorf("no host was found for the custom identifier: %s", customIdentifier)
+		logger.Infof("no host was found for the custom identifier: %s", customIdentifier)
+		return nil, err
```

Comment:
With this patch `mackerel.API.FindHostByCustomIdentifier` will not return (non-nil) `err` when a host with the CustomIdentifier is not found, but it should return no-nil error in order to the caller will retry the api request.
https://github.com/mackerelio/mackerel-agent/blob/f680106ca00d9292b7d62238991042780fca5053/command/command.go#L61-L64

https://github.com/mackerelio/mackerel-agent/blob/f680106ca00d9292b7d62238991042780fca5053/command/command.go#L41-L50